silkworm/node/db/access_layer.cpp

```diff
@@ -1222,10 +1222,13 @@ void write_runtime_states_u64(RWTxn& txn, uint64_t num, RuntimeState runtime_sta
 std::optional<eosevm::ConsensusParameters> read_consensus_parameters(ROTxn& txn, BlockNum index) {
     auto cursor = txn.ro_cursor(table::kConsensusParameters);
     auto key{db::block_key(index)};
-    auto data{cursor->find(to_slice(key), /*throw_notfound=*/false)};
+    auto data{cursor->lower_bound(to_slice(key), /*throw_notfound=*/false)};
```

Comment:
If we are going to do this way, we must make sure there's no garbage data in the table otherwise this search may return wrong values. 

But in case of forks, it is possible that we write wrong data in this table. Currently we do not have way to fix that at all.

e.g. it's possible that I received new config in block A, then this event is forked away.

Reply:
outdated now